nf_core/modules/module_test.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+
+"""from __future__ import print_function
+from rich.syntax import Syntax
+
+import errno
+import gzip
+import hashlib
+import operator
+
+
+import re
+import rich
+import shlex
+import subprocess
+import tempfile
+import yaml
+
+"""
+import nf_core.utils
+from .modules_repo import ModulesRepo
+from .test_yml_builder import ModulesTestYmlBuilder
+
+log = logging.getLogger(__name__)
+
+
+class ModulesTest(ModulesTestYmlBuilder):
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+    ):
+        self.run_tests = True
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.module_dir = None
+        self.module_test_main = None
+        self.entry_points = []
+        self.tests = []
+        self.errors = []
+
```

Comment:
I understand that. However, doing simple check in the beginning will not just make maintenance easier, it will also structure the code more nicely, separate concerns and this also makes the code easier to maintain. It would also give the net programmer a template how to change the other parts of the code for the better. Just to be clear, this comment was an attempt to nudge nf-core command files to be more structured, not to block this PR for not doing it.

Reply:
This particular style can be attributed to me as I've written most of the code base.

I've personally moved away from doing anything to inputs on instantiation. By using a dedicated function to do it, you give more flexibility to how the API is used - for example, creating an object and then assigning attributes subsequently before moving forward. If all attributes are checked immediately then this is impossible. So my personal style is generally not to "do" anything at all within `__init__()` - simply define the initial attributes and then handle other things later in separate functions.

This mostly comes from painful experiences within MultiQC where I have regretted using `__init__()` within classes to do stuff and have had to come up with weird and wonderful workarounds as a result.